### PR TITLE
Update the ref pack version to 6.0.0 for apicompat

### PR DESCRIPTION
**Pull request by Anipik**
Fixes https://github.com/dotnet/runtime/issues/59533

### CODE REVIEW

**Comment by safern** [comment]
It seems like we don't do this on other release branches. Is this because we are servicing the ref pack this time?

https://github.com/dotnet/runtime/blob/ddd0a9ca54bb6f851169c4ddd42cb4abf9448dcb/src/libraries/shims/ApiCompat.proj#L10

Also, we need to update this on main as we are running API Compat against 5.0.0 and not 6.0.0:
https://github.com/dotnet/runtime/blob/8c6aeea1d8e17fd692df0f509d415d9ad79d83bc/Directory.Build.props#L57

**Comment by Anipik** [comment]
> It seems like we don't do this on other release branches. Is this because we are servicing the ref pack this time?

yeah thats the reason

File: src/libraries/shims/ApiCompat.proj [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <BaselineApiCompat Condition="'$(BaselineApiCompat)' == ''">false</BaselineApiCompat>
     <PreviousNetCoreApp>net5.0</PreviousNetCoreApp>
```
Comment (safern) [inline src/libraries/shims/ApiCompat.proj:6]: It seems like we need to update this? We are hitting the error that checks the tfm exists in the ref pack we are comparing against.

**Comment by Anipik** [comment]
@safern can you approve this one ?

**Review by safern — APPROVED** [review]

**Pull request merged**